backup-tools/backup-replication-image/go-src/job_queue.go

```diff
@@ -0,0 +1,269 @@
+			if dlErr != nil {
+				PromGauge.SetError()
+				log.Printf("Worker%d File %d/%d (%s): Download File Error: %s\n", id, j.FileNumber, j.FileAllCount, j.File, dlErr)
+				// GoToEndWhileDownloadError used for exit the replication for error while downloading the file
+				goto GoToEndWhileDownloadError
+				break
+			}
+			lastDownload = true
+		}
+
+		// Upload file if no error before was triggered, and the dlFilePath is longer then 0
+		if dlErr == nil && len(dlFilePath) > 0 {
+			fakeName := strings.TrimPrefix(dlFilePath, backupDir)
+			if DebugOutput == "yes" {
+				log.Printf("Worker%d File %d/%d: Upload File: %s to %s\n", id, j.FileNumber, j.FileAllCount, fakeName, to.Cfg.OsRegionName)
+			}
+			uploadDone, upErr = swiftcli.SwiftUploadFile(to.SwiftCli, dlFilePath, &expiration, &fakeName)
+			if upErr != nil || !uploadDone {
+				PromGauge.SetError()
+				log.Printf("Worker%d File %d/%d (%s): Upload File Error: %s\n", id, j.FileNumber, j.FileAllCount, j.File, upErr)
+			}
+		}
+	}
+
+	// GoToEndWhileDownloadError used for exit the replication for error while downloading the file
+GoToEndWhileDownloadError:
+
+	if _, err := os.Stat(dlFilePath); !os.IsNotExist(err) {
+		os.Remove(dlFilePath)
+	}
+
+	currentFilesDone += 1
+
+	PromGauge.CurrentFile(currentFilesDone)
+
+	if DebugOutput == "yes" {
+		log.Printf("Worker%d File %d/%d: completed %s\n", id, j.FileNumber, j.FileAllCount, j.File)
+	}
+
+	num := int(Round((float64(currentFilesDone) / float64(j.FileAllCount)) * 100.0))
+
+	if num > alreadyPrinted || num == alreadyPrinted {
+		alreadyPrinted += 5
+		log.Printf("%d%% of replication done\n", num)
+	}
+
+}
+
+func StartJobWorkers() {
+
+	var fileCounter = 0
+	var countAllFiles = 0
+	var startTime = time.Now()
+	log.Println("Start replication task!")
+	// channel for jobs
+	jobs := make(chan Job)
+
+	// start workers
+	wg := &sync.WaitGroup{}
+	wg.Add(maxWorkers)
+	for i := 1; i <= maxWorkers; i++ {
+		go func(i int) {
+			defer wg.Done()
+
+			for j := range jobs {
+				doWork(i, j)
+			}
+		}(i)
+	}
+
+	for id, file := range EnvFrom.Files {
+
+		if stringInAllSlice(file, EnvTo) {
+			EnvFrom.Files[id] = ""
+			//log.Printf("Skip File: %s its already on all replication regions\n", file)
+		}
+		if strings.HasSuffix(file, "base") {
+			EnvFrom.Files[id] = ""
+		}
+
+	}
+
+	EnvFrom.Files = utils.DeleteEmpty(EnvFrom.Files)
+
+	countAllFiles = len(EnvFrom.Files)
+	PromGauge.AllFiles(countAllFiles)
+
+	// add jobs
+	for _, file := range EnvFrom.Files {
+
+		fileCounter += 1
+
+		jobs <- Job{
+			EnvFrom:      EnvFrom,
+			EnvTo:        EnvTo,
+			File:         file,
+			FileAllCount: countAllFiles,
+			FileNumber:   fileCounter,
+		}
+	}
+	close(jobs)
+
+	// wait for workers to complete
+	wg.Wait()
+
+	os.RemoveAll(backupDir)
+
+	var replicationDuration = time.Since(startTime)
+	log.Printf("End replication task in %v\n", replicationDuration-(replicationDuration%time.Second))
```

Comment:
Shorten this to `log.Printf("End replication task in %g seconds\n", time.Since(startTime).Seconds())`.